pipelines/translation.py

```diff
@@ -0,0 +1,262 @@
+from kfp import components, dsl
+from kubernetes.client import V1Toleration
+from kubernetes.client.models import V1PersistentVolumeClaimVolumeSource
+
+from . import training
+from .common import *
+
+
+@dsl.pipeline(name='Full multilingual pipeline for dialogue datasets', description='Generate, Translate, Train, and Eval')
+def generate_translate_train_eval_pipeline(
+    owner='',
+    project='',
+    experiment='',
+    s3_bucket='geniehai',
+    source='user',
+    input_splits='train eval',
+    model_name_or_path='Helsinki-NLP/opus-mt-$(src_lang)-$(tgt_lang)',
+    nmt='marian',
+    src_lang='en',
+    tgt_lang='',
+    model='',
+    dataset='',
+    image=default_image,
+    genienlp_version=GENIENLP_VERSION,
+    genie_version=GENIE_VERSION,
+    workdir_repo=WORKDIR_REPO,
+    workdir_version=WORKDIR_VERSION,
+    thingpedia_developer_key=default_developer_key,
+    translation_additional_args='--val_batch_size 2000 --temperature 0.2 --translate_example_split',
+    generate_dataset_parallel='6',
+    generate_dataset_additional_args='',
+    train_task_name='almond_dialogue_nlu',
+    train_load_from='None',
+    train_additional_args='',
+    train_iterations='20000',
+    valid_set='eval',
+    eval_set='dev',
+    eval_parallel_jobs='2',
+    eval_additional_args='',
+    fewshot_train_iterations='10000',
+):
+    training.everything(
+        do_generate=True,
+        do_translate=True,
+        do_bootleg=False,
+        do_paraphrase=False,
+        do_fewshot=False,
+        do_calibrate=False,
+        do_ood=False,
```

Comment:
Production models need bootleg, fewshot, calibration, ood. We need a pipeline with those on.
Btw, does bootleg work with non-English data? Do we need to train a Bootleg against non-English wikipedia?

In reality, I wonder if we should have a mega-pipeline that has all the languages we care about, with Generate (shared with English), and then one branch of each model for each language. It might be expensive though.

Reply:
It will be difficult to use for anything if it's not going to give us good accuracy, at least in one skill. We definitely need some way to validate on native language data, and some fewshot. This is going to be tricky.

Reply:
I don't trust testing on translated data. But anyway let's not worry about it for now. We need to bootstrap first.

Reply:
yeah so it's either user logs or crowd-sourcing to get real data.